### PR TITLE
Webpack

### PR DESCRIPTION
**Pull request by pgutkowski**
I know it can't automatically merge, I opened this request only to show You solution I found. Didn't want to waste time on resolving conflicts, check Your mailbox for explanation. 
Tests will surely fail, they are not configured to support import statements.

### CODE REVIEW

**Comment by rkorytkowski** [comment]
You are correct. I've checked more examples on the web (e.g. https://github.com/AngularClass/NG6-starter/tree/master/client/app) and all follow the pattern of declaring separate modules. I guess it is due to the fact that in order to use the conceptDictionaryApp module in a controller you would have to import its declaration from conceptDictionaryApp.js and we would end up with a circular dependency issue. Please see my comments in the pull request on how to name things and then resolve conflicts. I will fine tune webpack config a bit once you are done so that we have livereload and can get rid of bower and gulp.

File: app/js/classAdd.controller.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export default angular
-    .module('conceptDictionaryApp')
+    .module('stuff')
```
Comment (rkorytkowski) [inline app/js/classAdd.controller.js]: Separate modules should be created for each controller (a piece of functionality). Let's name it conceptDictionaryApp.classAdd. Apply the same pattern to other modules. The "conceptDictionaryApp." part indicates that it is a submodule of conceptDictionaryApp.

File: app/js/conceptDictionaryApp.module.js [hunk 2]
```diff
@@ -10,9 +13,9 @@ import DataTypesDetailsController from './dataTypesDetails.controller.js';
 import DataTypesList from './dataTypesList.controller.js';
 import conceptLocaleService from './conceptLocale.service.js'
 
-angular
+export default angular
     .module('conceptDictionaryApp',
-		['ngRoute', 'openmrs'])
+		['ngRoute', 'openmrs', 'stuff'])
```
Comment (rkorytkowski) [inline app/js/conceptDictionaryApp.module.js]: Declare dependencies on all submodules.

**Comment by pgutkowski** [comment]
@rkorytkowski ok, I'm on it. I will probably have to find way to configure tests, so it will pass Travis, right?

**Comment by rkorytkowski** [comment]
It was what I was planning to do as well, but you can give it a try if you want. If you are blocked, please push your progress and pass it to me :)

**Comment by rkorytkowski** [comment]
Just one comment. You will want to make 'npm run test' to run karma and configure travis to use npm instead of 'gulp test'.

**Comment by pgutkowski** [comment]
Rafał, I made changes You asked for and pushed them on my branch, but I couldn't properly pull --rebase nor pull the changes from Your master branch. Git gets it wrong at auto-merging, doesn't show all the conflicts and overwrites changes from my last commit. I think it has something to do with Tomek's commits from yesterday, when we tried to add webpack, and erased the changes in next commit. Do You know how can I solve this problem?

**Comment by rkorytkowski** [comment]
Let me try rebasing myself. I'll get back to you once I figure it out and how I did that.

**Comment by rkorytkowski** [comment]
Okay, so here's the recipe:
1. git rebase -i HEAD~4
2. Let's squash 4 commits with the webpack feature into 1 (3 yours and 1 Tomek's)
3. git pull --rebase upstream master

Basically we have reverted Tomek's changes in master so we need to reapply them again. The easiest way is to just squash his commit together with yours. That way git recognizes his changes as completely new ones and will reapply them.

I have already done that, fixing conflicts too so I will issue a new pull request of my branch and do testing. Thanks Pawel!

**Comment by pgutkowski** [comment]
Ok, that's great, thanks!

**Comment by rkorytkowski** [comment]
Please see https://github.com/rkorytkowski/openmrs-owa-conceptdictionary/pull/30

**Pull request closed without merging**